bin/docserv

```diff
@@ -42,7 +42,6 @@ class RepoLock:
         self.thread_id = thread_id
 
     def acquire(self, blocking=True):
-        logger.debug("Thread %i: Acquiring lock %s." % (self.thread_id, self.resource_name))
```

Comment:
Hmn... have you considered of rewriting RepoLock as a context manager? Or maybe even derive it from `threading.Lock`?  It makes it a bit easier to write and read as all the aquiring and releasing process is hidden inside the context manager.

However, not sure if this is really feasible here. For more information, see https://pymotw.com/3/threading/index.html#locks-as-context-managers.